ext/datadog_profiling_native_extension/heap_recorder.c

```diff
@@ -364,6 +376,8 @@ void heap_recorder_prepare_iteration(heap_recorder *heap_recorder) {
   if (heap_recorder->object_records_snapshot == NULL) {
     rb_raise(rb_eRuntimeError, "Failed to create heap snapshot.");
   }
+
+  return heap_recorder->iteration_gen;
```

Comment:
Minor: Given `iteration_gen` doesn't really change during iteration, may be easier to use `current_gc_gen` here (as otherwise the code slightly seems to suggest that they are different?)